### DISCUSSION TITLE
Support for ClusterClass

### DISCUSSION THREAD

**Discussion by prometherion** [clastix/cluster-api-control-plane-provider-kamaji#90] [Ideas]
The `ClusterTopology` Cluster API feature allows the creation of clusters based on a class.

This requires the Kamaji Control Plane provider to offer a new API resource named `KamajiControlPlaneTemplate`, which is a 1:1 copy of the specification (YMMV) with the removal of fields that would be useless when replicating a Control Plane (e.g.: IP address).

**Comment by hexchen** [comment]
As far as I can tell from briefly looking at the implementations of other providers as well as the documentation, implementing this might really only require providing a new CRD for the `KamajiControlPlaneTemplate` resource. I _think_ that the actual templating is done by the capi-controller-manager.

E.g. https://github.com/kubernetes-sigs/cluster-api/blob/main/controlplane/kubeadm/internal/webhooks/kubeadmcontrolplanetemplate.go only does some input validation and enforces the immutability of the `KubeadmControlPlaneTemplate`.

**Reply by chess-knight** [reply]
See https://github.com/clastix/cluster-api-control-plane-provider-kamaji/pull/107

**Comment by prometherion** [comment]
Implemented with #107, ready to get shipped with v0.10.0.

The discussion can be closed, any bug report must be filled via Issues, unless a new enhancement proposal is proposed.
